crates/optimism/consensus/src/lib.rs

```diff
@@ -37,19 +44,23 @@ pub use validation::validate_block_post_execution;
 ///
 /// Provides basic checks as outlined in the execution specs.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct OpBeaconConsensus {
+pub struct OpBeaconConsensus<P> {
     /// Configuration
     chain_spec: Arc<OpChainSpec>,
+    provider: P,
 }
 
-impl OpBeaconConsensus {
+impl<P> OpBeaconConsensus<P> {
     /// Create a new instance of [`OpBeaconConsensus`]
-    pub const fn new(chain_spec: Arc<OpChainSpec>) -> Self {
-        Self { chain_spec }
+    pub const fn new(chain_spec: Arc<OpChainSpec>, provider: P) -> Self {
+        Self { chain_spec, provider }
     }
 }
 
-impl FullConsensus<OpPrimitives> for OpBeaconConsensus {
+impl<P> FullConsensus<OpPrimitives> for OpBeaconConsensus<P>
+where
+    P: StateProviderFactory + fmt::Debug,
```

Comment:
Unsure how you prefer me to interface from `FullNodeTypes::Provider: FullProvider` (`FullProvider: StateProviderFactory + ..`) to `StorageRootProvider` @onbjerg ?

Reply:
for example adding trait bound `StorageRootProvider` doesn't work out of the box here, and would like input on how to approach it best
https://github.com/paradigmxyz/reth/blob/abaeb35fd17acd9705fc8b23ed52c8f4f40368f4/crates/optimism/payload/src/builder.rs#L351

Reply:
added the trait bound `Provider: fmt::Debug` just to the impl `Node<N> for OpNode`. was required since `Debug` is super trait of `HeaderValidator` (`Consensus` supertrait), now that `OpBeaconConsensus` has a generic.